File: app/src/main/java/com/turlir/abakcalc/CalculatorVisual.java
package com.turlir.abakcalc;

import com.turlir.calculator.converter.Visual;

public interface CalculatorVisual extends Visual {

    boolean selectionConstraint(int selStart, int selEnd, int length);

    int[] interceptSelection(int nowS, int nowE);

    int constraintStart();

    int constraintEnd();
}

File: app/src/main/kotlin/com/turlir/abakcalc/CalculatorVisual.kt
package com.turlir.abakcalc

import com.turlir.calculator.converter.Visual

interface CalculatorVisual : Visual {

    fun selectionConstraint(selStart: Int, selEnd: Int, length: Int): Boolean

    fun interceptSelection(nowS: Int, nowE: Int): IntArray

    fun constraintStart(): Int

    fun constraintEnd(): Int
}
